### ISSUE TITLE
Looking for some guidance on how to properly write mixed() (especially dealing with invalid input)

### ISSUE THREAD

**Issue by jcoveney-anchorzero** [jquense/yup#2010]
this is something I've run up against a couple of times...it isn't obvious to me what the "right" way is to write transformers in the context of mixed. This is what I've come up with for now, which seems to work but has a pretty major downside
```
Yup.mixed<WrappedBigInt>(
    (value): value is WrappedBigInt => value instanceof WrappedBigInt,
  )
    .transform((value, _raw, ctx) => {
      if (ctx.isType(value)) return value;
      if (!value) return value;
      try {
        return new WrappedAzBigInt(BigInt(value));
      } catch (error) {
        return null;
      }
    })
    .required();
```

the downside is that the error is a little nonsensical. I get `'value is a required field'`. It makes sense why this is the error, but it has nothing to do with the actual issue...say if you have the value 1.5.

what is the "best" pattern for handling errors in this case? I thought of something else, like...
```
class Sentinel {
  constructor(readonly error: string) {}

  toJSON() {
    return this.toString();
  }

  toString() {
    return this.error;
  }
}

Yup.mixed<WrappedBigInt>(
    (value): value is WrappedBigInt => value instanceof WrappedBigInt,
  )
    .transform((value, _raw, ctx) => {
      if (ctx.isType(value)) return value;
      if (!value) return value;
      try {
        return new WrappedAzBigInt(BigInt(value));
      } catch (error) {
        return new Sentinel(`${value} couldn't be safely cast to a BigInt`);
      }
    })
    .test('sentinel-is-error', (value) => !(value instanceof Sentinel)
    .required();
```
this seems to work! though it's a little more roundabout than I'd like...but maybe this is the sort of thing we have to do? I'm not quite sure, but would certainly appreciate any guidance.

also, I'm not quite sure how to get the error that it shows to be a bit nicer. as is, I'm getting something like
`'value must match the configured type. The validated value was: `"1.5 couldn\'t be converted safely to BigInt"` (cast from the value `1.5`).',`

which makes sense...it's getting the sentinel value and reporting that as the validated value. I've tried to use textContext.createError, but can't seem to get it to work, oddly enough...

**Comment by jquense** [comment]
Yeah this is expected. We actually recently made this more common bc folks generally get confused when null values throw type errors instead of required errors, so null is always considered a missing value over an invalid one. An invalid sentinel value is the right way to indicate "failed to coerce" or just return the original input in that case. You shouldn't have to add a special test tho, the instanceof check should be sufficient to throw a typeError

**Comment by jcoveney-anchorzero** [comment]
Ah see! thank you very much for the response. I wonder, would you be open to a PR which enabled something like the following?
```
Yup.mixed<WrappedBigInt>(
    (value): value is WrappedBigInt => value instanceof WrappedBigInt,
  )
    .transform((value, _raw, ctx) => {
      if (ctx.isType(value)) return value;
      if (!value) return value;
      try {
        return new WrappedAzBigInt(BigInt(value));
      } catch (error) {
        return ctx.createError({message: `${value} could not be cast to a bigint`});
      }
    })
    .required();
```
I'd be happy to use whatever syntax/machinery you see fit, and will try my hand at submitting a PR if you think this is something that would make sense for Yup
